core/domain/exp_jobs_one_off.py

```diff
@@ -678,10 +683,35 @@ def map(file_model):
                             content, mimetype='image/%s' % filetype)
                         yield (FILE_COPIED, 1)
 
-
     @staticmethod
     def reduce(status, values):
         if status == FILE_COPIED:
             yield (status, len(values))
         else:
             yield (status, values)
+
+
+class CreateVersionsOfImageJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating compressed versions of the images
+    of the exploration in the GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(exp_model):
+        # This job is allowed to run only in Production environment since it
```

Comment:
That's not true. The DEV_MODE check should be removed. The seam should be in core/platforms and in dev it would create three images in the datastore instead.

Reply:
Okay, sure.